api/queries_issue.go

```diff
@@ -246,8 +248,23 @@ func IssueList(client *Client, repo ghrepo.Interface, state string, labels []str
 	if mentionString != "" {
 		variables["mention"] = mentionString
 	}
+
 	if milestoneString != "" {
-		variables["milestone"] = milestoneString
+		milestones, err := RepoMilestones(client, repo)
+		if err != nil {
+			return nil, err
+		}
+
+		for i := range milestones {
```

Comment:
If we were unable to find a matching milestone, this should error out with something like: `no milestone found with title: %q`